libuserhelper.cpp

```diff
@@ -269,3 +300,33 @@ QString LibUserHelper::homeDir(uint uid)
 
     return rv;
 }
+
+QStringList LibUserHelper::groups(uint uid)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
+    if (!context) {
+        qCWarning(lcSUM) << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return QStringList();
+    }
+
+    QStringList rv;
+    struct lu_ent *ent = lu_ent_new();
+    if (lu_user_lookup_id(context, uid, ent, &error)) {
```

Comment:
If this 'error' gets set, where is it supposed to get freed? Presumably in else block?

Reply:
Added frees.